src/NHibernate/Engine/ISessionImplementor.cs

```diff
@@ -66,6 +66,15 @@ internal static IDisposable BeginProcess(this ISessionImplementor session)
 				: SessionIdLoggingContext.CreateOrNull(session.SessionId);
 		}
 
+		internal static bool IsProcessing(this ISessionImplementor session)
+		{
+			if (session == null)
+				return false;
+			return session is AbstractSessionImpl impl
```

Comment:
nit: i think this can be replaced by `return session is AbstractSessionImpl impl && impl.IsProcessing;` but up to you really.